include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -160,6 +175,25 @@ MGConstrainedDoFs::initialize(const DoFHandler<dim,spacedim> &dof,
 }
 
 
+template <int dim, int spacedim>
+inline
+void
+MGConstrainedDoFs::set_zero_boundary_dofs(const DoFHandler<dim,spacedim> &dof,
+                                          const std::set<types::boundary_id> &boundary_indicators,
+                                          const ComponentMask &component_mask)
+{
+  // allocate an IndexSet for each global level. Contents will be
+  // overwritten inside make_boundary_list.
+  const unsigned int n_levels = dof.get_triangulation().n_global_levels();
+  boundary_indices.resize(n_levels);
```

Comment:
this makes me wonder what happens to previous content of the array. Is it possible that there was previous content but `size() != n_levels`? I would expect that this indicates an invalid use of the object.